### PR TITLE
Alex k ino/triage list client list

### LINKED ISSUE #7: [Triage View] Triage List - Client List
**Is your feature request related to a problem? Please describe.**
The center of the triage view is the triage list, which shows the cases that are currently relevant to help surface information intended to provide support and avoid letting negative outcomes percolate.

**Describe the solution you'd like**
(This is the repeated list of ‘Markov, Julia’ elements shown in the current mocks)

- The layout will be the same for all cards in the list, even when the types / filter menus change
- Some strings in these cards will change slightly depending on the ‘Type’ menu item selected (‘Early Discharges’ in the mock-up)
- The circular progress bar will not always have the same denominator (e.g., sometimes a list item will have only 3 needs to ‘complete’, others will have 7 as shown in the mock-up)
- The dates shown are in American date format, MM/DD/YY

Interactive elements still to be defined here.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by AlexK-ino**
## Description of the change
Added:

- customized progress bar
- tooltip box
- customized checkbox
- button "Snooze"
- mock data with American format date

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #7 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 .eslintrc.js
-react-app-env.d.ts
\ No newline at end of file
+react-app-env.d.ts
+reportWebVitals.ts
```
Comment (Wmaileq) [inline .eslintignore:3]: Add the new line

File: public/index.html [hunk 2]
```diff
@@ -17,7 +17,7 @@
     <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
     <title>Case Triage Tool</title>
   </head>
-  <body>
+  <body style="margin: 0;">
```
Comment (Wmaileq) [inline public/index.html:20]: Use global styles in favor of inline styles

File: src/components/Checkbox.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  className?: string;
+  onChange?: (event: React.MouseEvent<HTMLInputElement>) => void;
+}
+
+const CheckboxContainer = styled.div<Props>`
+  border: 1px solid #90aeb5;
+  width: 16px;
+  height: 16px;
+  border-radius: 3px;
+  display: ${(props: Props) => (props.checked ? "flex" : "none")};
```
Comment (Wmaileq) [inline src/components/Checkbox.tsx:31]: Use `props destructuring`.

File: src/components/ClientCard.tsx [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+  ): void => {
+    event.stopPropagation();
+    changeUser({
+      ...user,
+      expanded: false,
+      snoozed: !user.snoozed,
+    });
+  };
+
+  const setProgressColor = () => {
```
Comment (Wmaileq) [inline src/components/ClientCard.tsx:211]: `getProgressColor`
Also, as an alternative, you could do it as `IIFE` or `useMemo` and name it just `progressColor`.

File: src/components/IndividualCards.tsx [hunk 5]
```diff
@@ -14,28 +14,43 @@
...
-    <PersonCard />
-    <PersonCard />
-    <PersonCard />
-    <PersonCard />
-    <PersonCard />
-    <PersonCard />
-  </Container>
+export const IndividualCards = () => {
+  const [users, setUsers] = useState(mockData);
+  const handleChangeUser = (val: User) => {
```
Comment (Wmaileq) [inline src/components/IndividualCards.tsx:44]: Maybe just `changeUser` or `setUser`, or `setUserData`?

File: src/components/ClientCard.tsx [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+}
+
+interface Props {
+  user: User;
+  changeUser: (user: User) => void;
+}
+
+const Box = styled.div`
+  display: block;
+`;
```
Comment (Wmaileq) [inline src/components/ClientCard.tsx:48]: Please, add new lines between function declarations and before return operator. It looks too dense now.

Also, maybe separate styles into a different file and name it `*.styles.tsx`, thoughts?

File: src/App.test.tsx [hunk 7]
```diff
@@ -16,7 +16,7 @@
 // =============================================================================
 import React from "react";
 import { render, screen } from "@testing-library/react";
-import App from "./App";
+import { App } from "./App";
```
Comment (Wmaileq) [inline src/App.test.tsx:19]: Prefer `default export`

File: src/components/ClientCard.tsx [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+  surname: string;
+  id: string;
+  supervision: string;
+  supervisionLevel: string;
+  date: string;
+  progressSteps: number;
+  progress: number;
+  snoozed: boolean;
+  checked: boolean;
+  expanded: boolean;
```
Comment (Wmaileq) [inline src/components/ClientCard.tsx:38]: Boolean constants should start with `is`, `isExanded`, `isChecked`

File: src/components/ClientCard.tsx [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+  }
+  .red-indicator {
+    color: #ba4f4f;
+  }
+`;
+const Snooze = styled.input.attrs({ type: "image" })`
+  outline: 0;
+`;
+export const ClientCard = ({ user, changeUser }: Props) => {
+  const handleChangeChecked = (
```
Comment (Wmaileq) [inline src/components/ClientCard.tsx:191]: What is the mission of `handle` keyword here? Maybe name these functions `toggleIsChecked`, `toggleIsExpanded`, ...?

**Review by Wmaileq — CHANGES_REQUESTED** [review]
Initial review

**Review by jessex — CHANGES_REQUESTED** [review]
I tried to start the app locally via `yarn start` to test this and it failed to compile (see screenshot below). Is that happening for you, @AlexK-ino? I know we had a similar issue on an earlier PR where @ohaibbq and I could not start it locally, so if you're not trying to run this locally via `yarn start`, you definitely should do so before issuing PRs.

File: .github/pull_request_template.md [hunk 10]
```diff
@@ -17,13 +17,13 @@
...
 
-These boxes should be checked by the submitter prior to merging:
+These boxes should be isChecked by the submitter prior to merging:
 
 - [ ] Manual testing against realistic data has been performed locally
 
 ### Code review
 
-These boxes should be checked by reviewers prior to merging:
+These boxes should be isChecked by reviewers prior to merging:
```
Comment (ohaibbq) [inline .github/pull_request_template.md:26]: I don't think you meant to find/replace this.

**Comment by ohaibbq** [comment]
Thanks, Alex! Its looking good. Some styling issues to fix up before I review the code.

Please reference the checklist with my attached annotated image.

- [ ] 1. The font color and the circle fill color is based on what % full the circle is. Please see the mocks for the appropriate colors for the various circle states. (0,1,2,3,4,5,6,7). The circle height should be as tall as the 2 lines of text to the right
- [ ] 2. There is too much whitespace between the circle and the first column.
- [ ] 3. There is too much whitespace between the two lines of text
- [ ] 4. The "tooltip" text does vertical alignment does not match the alignment specified in the mocks
- [ ] 5. The left side of the exclamation icon should align with the left side of the circle 
- [ ] 6. On smaller screen sizes, an odd left border appears on the list item
- [ ] 7. When hovering an ID link, the text falls out of left alignment with the line above
- [ ] 8. The tooltip is not activated via click. It should be shown by default and is persistent
- [ ] 9. The tooltip is missing the arrow icon on the right side
- [ ] 10. The bottom border list items needs to be consistent. There is currently an inconsistency based on whether or not item has a tooltip
- [ ] 11. Please allow for some dates to have an red indicator icon to the right hand side. This will signify that the date is past due
Client List

**Comment by AlexK-ino** [comment]
Hello @ohaibbq, @jessex!
Thanks for the comment, I corrected all the above, I'm not sure what, I understood point # 10 correctly, but I corrected the shadows.

Regarding the point, I initially did not understand its purpose and therefore flew out of my head, I apologize.

The problem with the build was due to an accidental prettier rule change. I also apologize for this incident. I will try not to make similar mistakes in the future.

**Review by ohaibbq — APPROVED** [review]
@AlexK-ino This looks so good, nice work!

Let's get this merged. Could you review #24 and start work on #23  and #25?

**Review by jessex — APPROVED** [review]
Looks great, @AlexK-ino! Looks like there are some merge conflicts but don't block on me.

**Comment by ohaibbq** [comment]
@AlexK-ino could you fix the merge conflicts and get this PR merged?

**Comment by AlexK-ino** [comment]
@ohaibbq, I fixed the conflicts, but unfortunately, I cannot merge branches, there is no access

**Pull request merged**